src/java/org/apache/cassandra/security/PEMBasedSslContextFactory.java

```diff
@@ -143,7 +168,19 @@ else if (!StringUtils.isEmpty(keystore_password) && !keyPassword.equals(keystore
     @Override
     public boolean hasKeystore()
     {
-        return maybeFileBasedPrivateKey ? keystoreFileExists() :
+        return maybeFileBasedPrivateKey ? keystoreFileExists(false) :
+               !StringUtils.isEmpty(pemEncodedKey);
+    }
+
+    /**
+     * Decides if this factory has an outbound keystore defined - key material specified in files or inline to the configuration.
+     *
+     * @return {@code true} if there is an outbound keystore defined; {@code false} otherwise
+     */
+    @Override
+    public boolean hasOutboundKeystore()
+    {
+        return maybeFileBasedOutboundPrivateKey ? keystoreFileExists(true) :
                !StringUtils.isEmpty(pemEncodedKey);
     }
```

Comment:
1. code style for ternary operator
2. not feel like using boolean to toggle inbound or outbound 

```suggestion
    public boolean hasOutboundKeystore()
    {
        return maybeFileBasedOutboundPrivateKey
               ? outboundKeystoreContext.hasKeystore()
               : !StringUtils.isEmpty(pemEncodedKey);
    }
```